### PR TITLE
Add a simple top-level README.md that briefly describes the libraries in grmtools.

### PR DESCRIPTION
**Pull request by ltratt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    `lex`), or at run-time (i.e. dynamically loading grammars without producing
+    `.rs` files).
+
+  * `lrpar`: a parser for LR grammars. Takes Yacc files and produces a parser
+    that can produce a parse tree. Can be used at compile-time (i.e. to produce
+    `.rs` files, in similar fashion to `lex`), or at run-time (i.e. dynamically
+    loading grammars without producing `.rs` files).
+
+  * `lrtable`: a library for producing LR parse tables from grammars. Mostly
+    intended for use by `lrpar`, but may be useful separately for those who wish
```
Comment (snim2) [inline README.md]: For use *with*?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Grammar and parsing libraries for Rust
+
+This repository contains four Rust libraries for operating on Context-Free
+Grammars (CFGs) and parsing. Each can be used independently, or individually.
```
[the remark this answers is not in the document]
Comment (ltratt) [reply README.md]: Agreed.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+The libraries are neither complete nor stable, but they may still be useful in
+their current form.
+
+  * `cfgrammar`: a library for loading in and manipulating CFGs. Currently only
```
[the remark this answers is not in the document]
Comment (ltratt) [reply README.md]: Agreed.

**Comment by snim2** [comment]
Ok, just a few minor comments.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  * `lrlex`: a basic lexer. It takes input files in a format similar, but
+    deliberately not identical, to `lex` and produces a simple lexer. Can be
+    used at compile-time (i.e. to produce `.rs` files, in similar fashion to
+    `lex`), or at run-time (i.e. dynamically loading grammars without producing
+    `.rs` files).
+
+  * `lrpar`: a parser for LR grammars. Takes Yacc files and produces a parser
+    that can produce a parse tree. Can be used at compile-time (i.e. to produce
+    `.rs` files, in similar fashion to `lex`), or at run-time (i.e. dynamically
```
[the remark this answers is not in the document]
Comment (ltratt) [reply README.md]: Yes.

File: README.md (hunk 1 above)
Comment (ltratt) [reply README.md]: In retrospect, this whole description is a bit misleading. I've tried a bigger fix.

**Comment by ltratt** [comment]
OK, hopefully that commit makes things a bit clearer.

**Comment by snim2** [comment]
OK, please squash

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**